runners/extensions-java/metrics/src/main/java/org/apache/beam/runners/extensions/metrics/MetricsHttpSink.java

```diff
@@ -69,7 +74,7 @@ public void writeMetrics(MetricQueryResults metricQueryResults) throws Exception
     }
     int responseCode = connection.getResponseCode();
     if (responseCode != 200) {
-      throw new HTTPException(responseCode);
+      throw new IOException("Expected OK 200 response but received: " + responseCode);
```

Comment:
Agreed as there is no suitable http exception in java.net regular package. But please change the message to add more context "Exception while writing metrics to MetricsHttpSink : expected HTTP 200 OK response but received "